sql/core/src/test/resources/sql-tests/results/postgreSQL/create_view.sql.out

```diff
@@ -803,61 +806,77 @@ Not allowed to create a permanent view `temporal4` by referencing a temporary vi
 
 
 -- !query 71
-CREATE TABLE tbl1 ( a int, b int) using parquet
+CREATE VIEW temporal5 AS SELECT * FROM t1 WHERE num IN (SELECT num FROM t1 WHERE EXISTS (SELECT 1 FROM tt))
 -- !query 71 schema
 struct<>
 -- !query 71 output
-
+org.apache.spark.sql.AnalysisException
+Not allowed to create a permanent view `temporal5` by referencing a temporary view `tt`;
```

Comment:
I think we should also provide the workaround in the error message. Instead just telling the users temp views are not allowed, we can say users can use permanent views in these cases?

cc @imback82 @cloud-fan

Reply:
We can add something like `Create a temporary view to reference temporary views; a permanent view can only reference permanent views`, but sounds a bit redundant with the error message?

Reply:
Permanent views can also reference to tables. 

There exists a general issue in our existing error messages. We do not have a message reference book. When users hit the unclear error messages, what they can do is to read the source code or try different options until it works. This is not natural to end users who have zero background about Spark SQL. We need to correct them one by one.

Reply:
Got it. I will do a follow-up PR.

Reply:
Note: this is the existing error message. This PR just extends the temp view check to subqueries. Agree that we can still improve the error message.

Reply:
Looks like this is already addressed by #26731.